packages/storybook/cypress/storybook/markdownEditorHeader.spec.js

```diff
@@ -0,0 +1,59 @@
+// https://www.cypress.io/blog/2020/02/12/working-with-iframes-in-cypress/
+const getIframeDocument = () =>
+  cy.get("#storybook-preview-iframe").its("0.contentDocument").should("exist");
+
+const getIframeBody = () =>
+  getIframeDocument().its("body").should("not.be.undefined").then(cy.wrap);
+
+describe(" Placing cursor in paragraph and changing to header 1, 2, 3", () => {
+    it("Change to Header 3", () => {
+      cy.visit("/");
+      //Finds the paragraph and place cursor
+      getIframeBody().find("#ap-rich-text-editor > p:nth-child(2)").click();
+      //Find heading dropdown and select heading-3
+      getIframeBody().find("#ap-rich-text-editor-toolbar > div.ui.simple.dropdown").click();
+      getIframeBody().find("#ap-rich-text-editor-toolbar > div.ui.active.visible.simple.dropdown > div.menu.transition.visible > div:nth-child(4)").click();
+      //checks if para changed to heading-3
+      getIframeBody().find("#This-is-text-This-is-italic-text-This-is-bold-text-This-is-a-undefined-This-is-inline-code").should('have.css', 'font-size', '16px', 'font-weight', 'bold');
+      //undo and check
+      getIframeBody().find("#ap-rich-text-editor-toolbar > svg:nth-child(11)").click();
+      getIframeBody().find("#ap-rich-text-editor > p:nth-child(2)").should('exist');
+      //redo and check
+      getIframeBody().find("#ap-rich-text-editor-toolbar > svg:nth-child(12)").click();
+      getIframeBody().find("#This-is-text-This-is-italic-text-This-is-bold-text-This-is-a-undefined-This-is-inline-code").should('have.css', 'font-size', '16px', 'font-weight', 'bold');
+     });
```

Comment:
Some initial thoughts... (I'll still do more looking into this whole PR)

- Maybe we make some constants at the top of the file to describe what a lot of these are? For example:
```js
const BUTTON_UNDO = 'svg:nth-child(11)';
const ID_TOOLBAR = '#ap-rich-text-editor-toolbar';
// ... etc
```
Could potential make this more readable.
- Similarly, could make a new line for each action:
```js
getIframeBody()
  .find("#ap-rich-text-editor > p:nth-child(2)")
  .should('exist');
```
- Not sure I understand the `// undo and check` part. When undoing, we check to see if the paragraph exists? Is this checking to see if it is a `p` and no longer has the heading properties?

Reply:
Yup I wanted to check whether it is a paragraph or not. Should I stick to css checking as I did for headers?

I'll make the code format changes and commit the code.

Thanks for the review @irmerk .